src/scripts/main.js

```diff
@@ -1,3 +1,38 @@
 'use strict';
 
-// write your code here
+const logo = document.querySelector('.logo');
+
+function handleClick() {
+  const promise1 = new Promise((resolve) => {
+    resolve('Promise was resolved!');
+  });
+
+  promise1.then(handleSuccess).catch(handleError);
+}
+
+function createRejectingPromise(message) {
+  return new Promise((resolve, reject) => {
+    setTimeout(() => {
+      reject(new Error(message));
+    }, 3000);
+  });
+}
+
+logo.addEventListener('click', handleClick);
+
+const promise2 = createRejectingPromise('Promise was rejected!');
+const messageDiv = document.createElement('div');
+
+document.body.appendChild(messageDiv);
+
+const handleSuccess = (message) => {
+  messageDiv.className = 'message';
+  messageDiv.textContent = message;
+};
+
+const handleError = (errorMessage) => {
+  messageDiv.className = 'message error-message';
+  messageDiv.textContent = errorMessage;
+};
```

Comment:
The handleError function should handle the error object itself, not just the error message. This will allow you to handle different types of errors in a more flexible way. For example, you can log the error stack trace, or handle specific types of errors differently.